charms/layers/jupyter-notebook/templates/config_template.py

```diff
@@ -0,0 +1,7 @@
+# Set options for certfile, ip, password, and toggle off
+# Set ip to '*' to bind on all interfaces (ips) for the public server
+c.NotebookApp.ip = '*'
+c.NotebookApp.open_browser = False
+
+# It is a good idea to set a known, fixed port for server access
+c.NotebookApp.port = {{port}}
```

Comment:
It's best practice to always end your file with a single newline. The red warning sign you see here indicates that there is no newline. Many editors, like Atom, automatically add a newline on saving the file.